packages/manager/modules/sms/src/sms/dashboard/creditTransfer/creditTransfer.controller.js

```diff
@@ -0,0 +1,74 @@
+export default class smsCreditTransferCtrl {
+  /* @ngInject */
+  constructor($translate, SmsService) {
+    this.$translate = $translate;
+    this.smsService = SmsService;
+  }
+
+  $onInit() {
+    this.loading = true;
+    this.model = {
+      serviceName: null,
+      smsAccountTarget: null,
+      credits: null,
+    };
+    this.maxCredit = null;
+
+    return this.smsService
+      .getAccountList()
+      .then((accounts) => {
+        this.accountsList = accounts.map((account) => ({
+          ...account,
+          displayName:
+            account.description !== ''
+              ? `${account.name} (${account.description})`
+              : account.name,
+        }));
+
+        this.model.serviceName = this.accountsList.find(
+          (account) => account.name === this.serviceName,
+        );
+        this.onSourceAccountChange(this.model.serviceName);
+      })
```

Comment:
nitpick: Can be moved to `resolve` in routing to have default error handling.

Reply:
I prefer to solve it in the controller because in the routing we cannot use the loader. We then have the impression that there is a time when nothing happens before the modal appears.